index.js

```diff
@@ -41,11 +48,21 @@ if (fs.existsSync("custom-name.js")) {
 
 if (fs.existsSync("stats.json")) {
 	statCache = JSON.parse(fs.readFileSync("stats.json"));
-	for (var key in statCache) {
-		if (!statCache.hasOwnProperty(key)) continue;
-		if (!statCache[key].mtimeSave) {delete statCache[key]; continue;}; //This code is just to load old stats.json file.
-		statCache[key].mtime = new Date(statCache[key].mtimeSave);
-	};
+	if (statCache.version != version) statCache = { version }; /* note: remember to change version every time stats.json format changes */
+	else {
+		for (let key in statCache) {
+			if (!statCache.hasOwnProperty(key) || key == "version") continue;
+			statCache[key].mtime = new Date(statCache[key].mtimeSave);
+		}
+	}
+}
+
+if (fs.existsSync("nonces.json")) {
+	nonces = JSON.parse(fs.readFileSync("nonces.json"));
```

Comment:
Wouldn't it be better if this was in a database or something? This sounds like a job for Redis.

Reply:
redis doesn't have persistence by default - the whole point of this is persistence, on the real shitty.download instance it takes about 3 minutes to fs.stat all the files